uint/src/uint.rs

```diff
@@ -1165,10 +1328,11 @@ macro_rules! construct_uint {
 			}
 		}
 
-		impl ::core::ops::Shl<usize> for $name {
+		impl<T> ::core::ops::Shl<T> for $name where T: Into<$name> {
```

Comment:
Seems a bit weird, you cannot really shift left that much. It should probably `panic` in case given argument is greater than `usize::max_value()` (currently it seems it will panic if it's greater than u64)

Reply:
the reason I did this was so we can << using any number/bignum type, not for allowing us to shift a very large amount

Reply:
I see. I would get rid of unsafe casting to `usize` anyway and panic explicitly.

Reply:
Addressed in new commit